### PR TITLE
[oracle] Fix missing sys metrics (DBMON-3191)

### PR DESCRIPTION
**Pull request by nenadnoveljic**
### What does this PR do?

Bug fix for missing sys metrics.

### Additional Notes

The `v$sysmetric` can include metrics from two time intervals, and this switch between intervals is not instantaneous. In other words, some metrics might belong to a previous time interval, and there could be instances where a metric is present in both intervals. The query sorts metrics based on their update date in descending order.

In the previous version, the loop would halt upon encountering any metric that had already been emitted. The recent modification ensures that the program continues iterating over all metrics, guaranteeing that metrics are emitted, even if they are not exclusively from the most recent interval.

### Describe how to test/QA your changes

Test if `shared_pool_free` metrics is being emitted to make sure that the change didn't break the existing funcitonality.

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

**Review by estherk15 — APPROVED** [review]

File: pkg/collector/corechecks/oracle-dbm/sysmetrics.go [hunk 1]
```diff
@@ -125,14 +125,16 @@ func (c *Check) sendMetric(s sender.Sender, r SysmetricsRowDB, seen map[string]b
...
 			s.Gauge(fmt.Sprintf("%s.%s", common.IntegrationName, metric.DDmetric), value, "", appendPDBTag(c.tags, r.PdbName))
 			seen[r.MetricName] = true
+			*n++
 		}
 	}
 }
 
-func (c *Check) SysMetrics() error {
+func (c *Check) SysMetrics() (int64, error) {
+	var n int64
```
Comment (lu-zhengda) [inline pkg/collector/corechecks/oracle-dbm/sysmetrics.go:134]: What's the use of `n`? is it an internal counter?

Comment (nenadnoveljic) [reply pkg/collector/corechecks/oracle-dbm/sysmetrics.go:134]: Yes, it's an internal counter of the emitted sys metrics. I exposed it in the return value for tests.

Comment (lu-zhengda) [reply pkg/collector/corechecks/oracle-dbm/sysmetrics.go:134]: ok. can we add a docstr to this function? it was not very clear to me that the counter is only used for tests, especially we ignored the counter [here](https://github.com/DataDog/datadog-agent/pull/21031/files#diff-c4f29b606f87d4700c343a33d1ebc82e5e85fbeb8697c61034c307056879678dR200).

Comment (nenadnoveljic) [reply pkg/collector/corechecks/oracle-dbm/sysmetrics.go:136]: done

**Review by lu-zhengda — APPROVED** [review]

**Pull request merged**